app/src/test/java/com/duckduckgo/mobile/android/duckduckgo/util/AppUrlsTest.java

```diff
@@ -22,75 +22,93 @@ public void whenGetHomeThenReturnHomeUrl() {
         final boolean result = AppUrls.isDuckDuckGo(url);
         assertFalse(result);
     }
     @Test
-    public void whenIsDuckDuckGoWithNonDDGValidUrlThatContainsDDGThenFails() {
+    public void whenUrlIsValidNotDDGAndContainsDuckDuckGoUrlThenIsDuckDuckGoIsFalse() {
         final String url = "https://www.test.com/?ko=-1&kl=wt-wt&q=duckduckgo.com";
         final boolean result = AppUrls.isDuckDuckGo(url);
         assertFalse(result);
     }
     @Test
-    public void whenIsDuckDuckGoWithRandomTextThenfails() {
+    public void whenUrlIsSimpleTextThenIsDuckDuckGoIsFalse() {
         final String url ="asd.oiasud";
         final boolean result = AppUrls.isDuckDuckGo(url);
         assertFalse(result);
     }
     @Test
-    public void whenIsDuckDuckGoWithRandomTextThatContainsDDGThenFails() {
+    public void whenUrlIsSimpleTextThatContainsDuckDuckGoThenIsDuckDuckGoReturnFalse() {
         final String url = "as///d.duckduckgo.com";
         final boolean result = AppUrls.isDuckDuckGo(url);
         assertFalse(result);
     }
     @Test
-    public void whenGetQueryWithDDGUrlFromWebWithQueryThenReturnCorrectQuery() {
+    public void whenUrlIsDDGFromWebWithQueryThenGetQueryReturnCorrectQuery() {
         final String url = "https://duckduckgo.com/?q=some+search&t=hc&ia=web";
         final String expected = "some search";
         final String result = AppUrls.getQuery(url);
         assertEquals(expected, result);
     }
     @Test
-    public void whenGetQueryWithDDGUrlWithQueryThenReturnCorrectQuery() {
+    public void whenUrlIsDDGWithQueryThenGetQueryReturnCorrectQuery() {
         final String url = "https://www.duckduckgo.com/?ko=-1&kl=wt-wt&q=some+search";
         final String expected = "some search";
         final String result = AppUrls.getQuery(url);
         assertEquals(expected, result);
     }
     @Test
-    public void whenGetQueryWithDDGUrlWithoutQueryThenReturnEmptyString() {
+    public void whenUrlIsDDGWithoutQueryThenGetQueryReturnEmptyString() {
         final String url = "https://www.duckduckgo.com/?ko=-1&kl=wt-wt";
         final String expected = "";
         final String result = AppUrls.getQuery(url);
         assertEquals(expected, result);
     }
     @Test
-    public void whenGetQueryWithNonDDGUrlTheReturnNull() {
+    public void whenUrlIsNotDDGWithQueryThenGetQueryReturnNull() {
         final String url = "https://www.test.com/?ko=-1&kl=wt-wt&q=some+search";
         final String result = AppUrls.getQuery(url);
         assertNull(result);
     }
     @Test
-    public void whenGetQueryWithRandomTextThenReturnNull() {
+    public void whenUrlIsRandomTextThenGetQueryReturnNull() {
         final String url = "asd''asd/////.com/?q=duckduckgo.com";
         final String result = AppUrls.getQuery(url);
         assertNull(result);
     }
     @Test
+    public void whenUrlIdDDGWithQueryWithEncodedCharactersThenGetQueryReturnCorrectQuery() {
+        final String[] urls = new String[] {
```

Comment:
yeah I agree, I though it was easier to read in a loop, but actually isn't, I'll change that now